oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/ClaimAccessor.java

```diff
@@ -49,7 +46,7 @@
 	 */
 	default Boolean containsClaim(String claim) {
```

Comment:
Now that these are converted up front, what are your thoughts on adding a method:

```java
default <T> T getClaim(String claim)
```

It'd be really nice for users to be able to do something like:

```java
List<String> scopes = jwt.getClaim("scope");
```

Reply:
I'm not sure how we would implement it with the latest implementation that uses a `ClaimConversionService`. We would need to supply the `targetType` using `ClaimConversionService.getSharedInstance().convert(claimValue, targetType)`, which must be captured via the return type of caller. Off the top of my head, I'm not aware how we can capture the generic return type? If it's even possible?

Either way, this would be a feature enhancement as it's not required for this PR. Feel free to log a ticket if you would like this enhancement.

Reply:
@jgrandja I think this can be a separate issue. However, I agree that this would be a nice feature.

Correct me if I'm wrong @jzheaux but I think the idea would be that it presumes it is already converted to the type. No conversion is done when `getClaim` is invoked.

Reply:
Agreed that it can be a separate issue. Yes, @rwinch this would simply do:

```java
default <T> T getClaim(String claim) {
    return (T) this.getClaims().get(claim);
}
```

Reply:
I now understand what you are suggesting @jzheaux. As @rwinch mentioned

> the idea would be that it presumes it is already converted to the type

I agree that this would be a nice improvement as long as the user is aware that the claim has already been converted otherwise a `ClassCastException` will occur.

I'll add a new ticket for this.

Reply:
See #6947